tests: uart_async_api: pic32cx_sg41_cult

### Describe the bug

It seems the changes which were made with the PR https://github.com/zephyrproject-rtos/zephyr/pull/108994/ should also be made in `tests/drivers/uart/uart_async_api` the Twister Tests 5 & 6 are failing.

See here:

- https://github.com/zephyrproject-rtos/zephyr/actions/runs/25775572155/job/75731994323?pr=109015
- https://github.com/zephyrproject-rtos/zephyr/actions/runs/25779382997/job/75718983633?pr=106687
- https://github.com/zephyrproject-rtos/zephyr/actions/runs/25777012062/job/75712050583?pr=105366

### Steps to reproduce

Create a PR and Run the Twister Tests

### Relevant log output

```shell
INFO    - 193/965 pic32cx_sg41_cult/pic32cx1025sg41128 drivers.uart.async_api.mchp             ERROR CMake build failure (build <zephyr/gnu>)
INFO    - /__w/zephyr/zephyr/twister-out/pic32cx_sg41_cult_pic32cx1025sg41128/zephyr_gnu/tests/drivers/uart/uart_async_api/drivers.uart.async_api.mchp/build.log
ERROR   - Loading Zephyr module(s) (Zephyr base): zephyr_default:kconfig
-- Application: /__w/zephyr/zephyr/tests/drivers/uart/uart_async_api
-- CMake version: 3.28.3
-- Found Python3: /opt/python/venv/bin/python3 (found suitable version "3.12.3", minimum required is "3.12") found components: Interpreter 
-- Cache files will be written to: /root/.cache/zephyr
-- Zephyr version: 4.4.99 (/__w/zephyr/zephyr)
-- Found west (found suitable version "1.5.0", minimum required is "0.14.0")
-- Board: pic32cx_sg41_cult, qualifiers: pic32cx1025sg41128
-- Found host-tools: zephyr 1.0.1 (/opt/toolchains/zephyr-sdk-1.0.1)
-- Found toolchain: zephyr 1.0.1 (/opt/toolchains/zephyr-sdk-1.0.1)
-- Found Dtc: /opt/toolchains/zephyr-sdk-1.0.1/hosttools/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.7.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/microchip/pic32c/pic32cx_sg41_cult/pic32cx_sg41_cult.dts
-- Found devicetree overlay: /__w/zephyr/zephyr/tests/drivers/uart/uart_async_api/boards/pic32cx_sg41_cult.overlay
devicetree error: 'dipo' appears in /soc/sercom@40003000 in /__w/zephyr/zephyr/twister-out/pic32cx_sg41_cult_pic32cx1025sg41128/zephyr_gnu/tests/drivers/uart/uart_async_api/drivers.uart.async_api.mchp/zephyr/zephyr.dts.pre, but is not declared in 'properties:' in /__w/zephyr/zephyr/dts/bindings/serial/microchip,sercom-g1-uart.yaml
CMake Error at cmake/modules/dts.cmake:324 (execute_process):
  execute_process failed command indexes:
    1: "Child return code: 1"
Call Stack (most recent call first):
  cmake/modules/dts.cmake:438 (dts_edt_pickle)
  cmake/modules/zephyr_default.cmake:133 (dts_init)
  cmake/modules/zephyr_default.cmake:133 (cmake_language)
  share/zephyr-package/cmake/ZephyrConfig.cmake:75 (include)
  share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  cmake/package_helper.cmake:126 (find_package)
```

### Impact

Showstopper – Prevents release or major functionality; system unusable.

--- a/tests/drivers/uart/uart_async_api/boards/pic32cx_sg41_cult.overlay
+++ b/tests/drivers/uart/uart_async_api/boards/pic32cx_sg41_cult.overlay
@@ -10,6 +10,11 @@ dut: &sercom0 {
 	#size-cells = <0>;
 	current-speed = <115200>;
 
+	/* sercom0 is configured for SPI, drop SPI-related properties to allow UART use */
+	/delete-property/ dipo;
+	/delete-property/ dopo;
+	/delete-property/ cs-gpios;
+
 	/* Internally loop-back TX and RX on PAD0 */
 	rxpo = <0>;
 	txpo = <0>;

